packages/squiggle-lang/src/rescript/Utility/E/E_A.res

```diff
@@ -305,56 +305,50 @@ module Floats = {
     /*
       This function goes through a sorted array and divides it into two different clusters:
       continuous samples and discrete samples. The discrete samples are stored in a mutable map.
-      Samples are thought to be discrete if they have any duplicates.
+      Samples are thought to be discrete if they have at least `minDiscreteWight` duplicates.
+
+      If the min discreet weight is 4, that would mean that at least four elements needed from a specific
+      value for that to be kept as discrete. This is important because in some cases, we can expect that
+      some common elements will be generated by regular operations. The final continous array will be sorted.
  */
-    let _splitContinuousAndDiscreteForDuplicates = (sortedArray: array<float>) => {
+    let splitContinuousAndDiscreteForMinWeight = (
+      sortedArray: array<float>,
+      ~minDiscreteWeight: int,
+    ) => {
       let continuous: array<float> = []
       let discrete = FloatFloatMap.empty()
-      Belt.Array.forEachWithIndex(sortedArray, (index, element) => {
-        let maxIndex = (sortedArray |> Array.length) - 1
-        let possiblySimilarElements = switch index {
-        | 0 => [index + 1]
-        | n if n == maxIndex => [index - 1]
-        | _ => [index - 1, index + 1]
-        } |> Belt.Array.map(_, r => sortedArray[r])
-        let hasSimilarElement = Belt.Array.some(possiblySimilarElements, r => r == element)
-        hasSimilarElement
-          ? FloatFloatMap.increment(element, discrete)
-          : {
-              let _ = Js.Array.push(element, continuous)
+
+      let flush = (cnt: int, value: float): unit => {
```

Comment:
Is `cnt` meant to mean `count`? I really don't recommending abbreviating things so heavily, it's not clear to me.

Reply:
Related, ``flush`` doesn't sound very meaningful; I recommend something more semantic, like, ``addDiscreteOrContinuousData``

Reply:
How do you feel about `flushBuffer`? I think in this case it's important to translate the algorithmic intent, i.e. "we collect the values and then eventually dump them into one of two targets".

`addData` also seems good to me, but I don't think `DiscreteOrContinuous` adds any value here, since the entire function is about DiscreteOrContinuous.

Reply:
That works